platform/admin/configurator/admin-configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/PropertyOperation.java

```diff
@@ -0,0 +1,206 @@
+
+    /**
+     * Transactional handler for deleting property files
+     */
+    private static class DeleteHandler extends PropertyOperation {
+        private DeleteHandler(Path configFile) {
+            super(configFile, Collections.emptyMap(), true);
+        }
+
+        @Override
+        public Result<Void> commit() throws ConfiguratorException {
+            boolean delete = configFile.delete();
+            if (!delete) {
+                LOGGER.debug("Problem deleting properties file {}", configFile);
+            }
+
+            return ResultImpl.pass();
+        }
+    }
+
+    /**
+     * Transactional handler for updating property files
+     */
+    private static class UpdateHandler extends PropertyOperation {
+        private final boolean keepIgnored;
+
+        UpdateHandler(Path configFile, Map<String, String> configs, boolean keepIgnored) {
+            super(configFile, configs, true);
+
+            this.keepIgnored = keepIgnored;
+        }
+
+        @Override
+        public Result<Void> commit() throws ConfiguratorException {
+            Map<String, String> propertyMap = new HashMap<>();
+
+            if (keepIgnored) {
+                propertyMap.putAll(currentProperties);
+            }
+            propertyMap.putAll(configs);
+            saveProperties(propertyMap);
+
+            return ResultImpl.pass();
+        }
+    }
+
+    final File configFile;
+
+    protected final Map<String, String> configs;
+
+    final Map<String, String> currentProperties;
+
+    private PropertyOperation(Path configFile, Map<String, String> configs,
+            boolean loadCurrentProps) {
+        this.configFile = configFile.toFile();
+        this.configs = new HashMap<>(configs);
+
+        currentProperties = new HashMap<>();
+        if (loadCurrentProps) {
+            Properties result;
+            try {
+                Properties props = new Properties();
+                try (FileInputStream in = new FileInputStream(configFile.toFile())) {
+                    props.load(in);
+                }
+
+                result = props;
+            } catch (IOException e) {
+                throw new ConfiguratorException(String.format(
+                        "Error reading configuration from file %s",
+                        configFile.toFile()
+                                .getName()));
+            }
+
+            result.keySet()
+                    .stream()
+                    .map(String.class::cast)
+                    .forEach(k -> currentProperties.put(k, result.getProperty(k)));
+        }
+    }
+
+    @Override
+    public Result<Void> rollback() throws ConfiguratorException {
+        saveProperties(currentProperties);
+
+        return ResultImpl.rollback();
+    }
+
+    Map<String, String> readState() throws ConfiguratorException {
+        return Collections.unmodifiableMap(currentProperties);
+    }
+
+    void saveProperties(Map<String, String> propertyMap) throws ConfiguratorException {
```

Comment:
The nested classes are static. They don't have access to the private non-static methods. Java scoping rules are fun and this one befuddled me for a second.

It's a bit twisty in there, as the `PropertyOperation.Actions.create` method can happily invoke the private ctor of `PropertyOperation.CreateHandler` because it too is a static class but can't access `saveProperties` because it's in the outer class.